### ISSUE TITLE
Database architecture

### ISSUE THREAD

**Issue by ferncabrera** [ferncabrera/open_ims#33]
We need our database to be mapped on our ERD page. (Please add more info and links to this item)!

**Comment by ferncabrera** [comment]
Hey @kellygibson - Just wanted to share [this documentation]https://node-postgres.com/features/types) you and pin it to this ticket for ref. It is the package we are using to help us connect and communicate to our Postgres DB in Node/JavaScript. Not sure how much you can get from it but the documentation is quite concise and you can skim it to see the sort of querying we are going to be doing, you'll see there is a lot of saw SQL involved lol.

**Comment by Polsander** [comment]
Hi @kellygibson I am working on our first database implementation (Yay!). I will be storing users into our user table when they sign up. I am not sure if I remember there being a generated id or not, so if you could send me the link to your Lucid chart or any finalized database mock-ups I'd greatly appreciate it.

I will do some guess work for now, but this can always be changed once I get the required info :)

**Comment by kellygibson** [comment]
Hey Oliver, [here](https://lucid.app/lucidchart/f5c3a95e-afbb-450c-964f-dc3169a58ff6/edit?viewport_loc=-93%2C-735%2C3297%2C1885%2C0_0&invitationId=inv_570e2e6b-f018-41bc-b56b-ad6743f0f230) is the link to my lucid chart. You can also revisit the ER diagram in figma on the page named "References". If you wanna mess around with tables [here](https://docs.google.com/document/d/1ShQq-8j65L7E3To0LGsKCBD_L5l4QGS8VSaoZL3tI78/edit?usp=sharing)'s the SQL commands I started working on for making the tables.

**Comment by Polsander** [comment]
Thanks @kellygibson. I made your implementations, they look great and were so easy to copy and paste! LOL

2 things however:
1. We hash our passwords using blowfish cipher for password security. So password `monkey123` turns into gibberish --> `djfl;j412kl3j4`. This gibberish can be up to 74 characters, so I opted to make the VarChar for the password field 128 as opposed to 64.
2. I removed not null from the phone number since that isn't being asked for on the registration. This is something the user can include once they've already signed up. Additionally given that two people working may be in the same office/family, are we sure we want to make the phone number field unique?

Let me know, and thanks!
